mpu6050/mpu6050.go

```diff
@@ -1,95 +1,232 @@
-	return
+// you would get a value close to 6.3 radians (360 degrees).
+func (d *Device) AngularVelocity() (gx, gy, gz int32) {
+	const angvelOffset = 8
+	_ = d.RawData[angvelOffset+5] // This line fails to compile if RawData is too short.
+	gx = int32(convertWord(d.RawData[angvelOffset+0:])) * 4363 / 8192 * d.gRange
+	gy = int32(convertWord(d.RawData[angvelOffset+2:])) * 4363 / 8192 * d.gRange
+	gz = int32(convertWord(d.RawData[angvelOffset+4:])) * 4363 / 8192 * d.gRange
+	return gx, gy, gz
 }
 
-// SetClockSource allows the user to configure the clock source.
-func (d Device) SetClockSource(source uint8) error {
-	return d.bus.WriteRegister(uint8(d.Address), PWR_MGMT_1, []uint8{source})
+// Temperature returns the temperature of the device in milli-centigrade.
+func (d *Device) Temperature() (Celsius int32) {
+	const tempOffset = 6
+	return 1506*int32(convertWord(d.RawData[tempOffset:]))/512 + 37*1000
+}
+
+func convertWord(buf []byte) int16 {
+	return int16(binary.BigEndian.Uint16(buf))
+}
+
+// SetSampleRate sets the sample rate for the FIFO,
+// register ouput and DMP. The sample rate is determined
+// by:
+//
+//	SR = Gyroscope Output Rate / (1 + srDiv)
+//
+// The Gyroscope Output Rate is 8kHz when the DLPF is
+// disabled and 1kHz otherwise. The maximum sample rate
+// for the accelerometer is 1kHz, if a higher sample rate
+// is chosen, the same accelerometer sample will be output.
+func (p *Device) SetSampleRate(srDiv byte) (err error) {
+	// setSampleRate
+	var sr [1]byte
+	sr[0] = srDiv
+	if err = p.write8(_SMPRT_DIV, sr[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetClockSource configures the source of the clock
+// for the peripheral.
+func (p *Device) SetClockSource(clkSel byte) (err error) {
+	// setClockSource
+	var pwrMgt [1]byte
+
+	if err = p.read(_PWR_MGMT_1, pwrMgt[:]); err != nil {
+		return err
+	}
+	pwrMgt[0] = (pwrMgt[0] & (^_CLK_SEL_MASK)) | clkSel // Escribo solo el campo de clk_sel
+	if err = p.write8(_PWR_MGMT_1, pwrMgt[0]); err != nil {
+		return err
+	}
+	return nil
+}
+
+// SetRangeGyro configures the full scale range of the gyroscope.
+// It has four possible values +- 250°/s, 500°/s, 1000°/s, 2000°/s.
+// The function takes values of gyroRange from 0-3 where 0 means the
+// lowest FSR (250°/s) and 3 is the highest FSR (2000°/s).
+func (p *Device) SetRangeGyro(gyroRange byte) (err error) {
+	switch gyroRange {
+	case GYRO_RANGE_250:
+		p.gRange = 250
+	case GYRO_RANGE_500:
+		p.gRange = 500
+	case GYRO_RANGE_1000:
+		p.gRange = 1000
+	case GYRO_RANGE_2000:
+		p.gRange = 2000
+	default:
+		return errors.New("invalid gyroscope FSR input")
+	}
+	// setFullScaleGyroRange
+	var gConfig [1]byte
+
+	if err = p.read(_GYRO_CONFIG, gConfig[:]); err != nil {
+		return err
+	}
+	gConfig[0] = (gConfig[0] & (^_G_FS_SEL)) | (gyroRange << _G_FS_SHIFT) // Escribo solo el campo de FS_sel
+
+	if err = p.write8(_GYRO_CONFIG, gConfig[0]); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
Can you make `gyroRange` a type? Like is done in the LIS3DH driver:

https://github.com/tinygo-org/drivers/blob/820298bfe3e9f8192e6c7c1147b95d7eee02eb46/lis3dh/lis3dh.go#L62

https://github.com/tinygo-org/drivers/blob/820298bfe3e9f8192e6c7c1147b95d7eee02eb46/lis3dh/registers.go#L54-L61

This makes it possible to easily change the type at a later time, if needed.